### PR TITLE
Issue : #[1070](https://github.com/ContainX/openstack4j/issues/1070)

### PR DESCRIPTION
**Pull request by kushalagrawal1**
Description: 1. Updated Apache httpclient version and corresponding code for IPv6+ https issue.
             2. Added support for IgnoreSSLVerification and custom SSLHostnameVerifier.

### CODE REVIEW

**Comment by vinodborole** [comment]
LGTM

**Comment by kushalagrawal1** [comment]
@gondor @auhlig can you guys please check and approve the pull request.

**Comment by auhlig** [comment]
Will take a look tomorrow.

File: connectors/resteasy/pom.xml [hunk 1]
```diff
@@ -19,6 +19,12 @@
 			<groupId>org.jboss.resteasy</groupId>
 			<artifactId>resteasy-jaxrs</artifactId>
 			<version>${resteasy-version}</version>
+			<exclusions>
+				<exclusion>
+					<artifactId>httpclient</artifactId>
+					<groupId>org.apache.httpcomponents</groupId>
+				</exclusion>
+			</exclusions>
```
Comment (auhlig) [inline connectors/resteasy/pom.xml]: Why excluding the httpclient from resteasy?

Comment (kushalagrawal1) [inline connectors/resteasy/pom.xml]: We tried using higher version of resteasy jar. But it did not work because of the version resteasy usage of httpclient. There are few SSL related bug fix only in 4.5.3 version.
https://archive.apache.org/dist/httpcomponents/httpclient/RELEASE_NOTES-4.5.x.txt

Comment (auhlig) [inline connectors/resteasy/pom.xml]: Which version of resteasy did you try?

Comment (kushalagrawal1) [inline connectors/resteasy/pom.xml]: I used 3.1.3.Final. now I can see that they have released new version (3.1.4.Final).   I will try to use this version and will upgrade if it works. thanks

Comment (kushalagrawal1) [inline connectors/resteasy/pom.xml]: Hi @auhlig , Even new one also come with httpcomponents:httpclient:jar:4.5.2

[INFO] ------------------------------------------------------------------------
[INFO] Building OpenStack4j RestEasy Connector 3.0.5-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- maven-dependency-plugin:2.8:tree (default-cli) @ openstack4j-resteasy ---
[INFO] org.pacesys.openstack4j.connectors:openstack4j-resteasy:jar:3.0.5-SNAPSHOT
**[INFO] +- org.jboss.resteasy:resteasy-jaxrs:jar:3.1.4.Final:compile**
[INFO] |  +- org.jboss.spec.javax.ws.rs:jboss-jaxrs-api_2.0_spec:jar:1.0.1.Beta1:compile
[INFO] |  +- org.jboss.resteasy:resteasy-jaxrs-services:jar:3.1.4.Final:compile
[INFO] |  +- org.jboss.spec.javax.annotation:jboss-annotations-api_1.2_spec:jar:1.0.0.Final:compile
[INFO] |  +- javax.activation:activation:jar:1.1.1:compile
**[INFO] |  +- org.apache.httpcomponents:httpclient:jar:4.5.2:compile**

Comment (auhlig) [inline connectors/resteasy/pom.xml]: Sorry for the late response and thanks for investigating @kushalagrawal1. Too bad it's not done just with resteasy.
Since you already updated resteasy on your system, do you mind adding another commit with that?

Comment (kushalagrawal1) [inline connectors/resteasy/pom.xml]: Hi @auhlig, I have not yet changed the code according to resteasy-3.x  As there is client creation process has been in latest 3.x  it will require the corresponding client creation code to be changed.

_when we are planning to release 3.0.5 version for OS4j ? 
 _I will try to push that change before that so this pull request can go along.__

Comment (auhlig) [inline connectors/resteasy/pom.xml]: Ah. I thought you started coding for the resteasy 3.x and I didn't want to get the work lost.
Else this could also live in a separate PR - as you like. 

On the issue here: Everyone okay with the workaround? @vinodborole 

No specific date. Release is currently done when sufficient changes piled up. I guess it would be justified to release soon.

Comment (vinodborole) [inline connectors/resteasy/pom.xml]: LGTM

Comment (kushalagrawal1) [inline connectors/resteasy/pom.xml]: @auhlig @vinodborole I will try to finish it by the weekend. If you release it before, then you can merge this pull request and leave a  comment. I will create a separate issue and commit accordingly.

Comment (auhlig) [inline connectors/resteasy/pom.xml]: No need to rush the release :)

Comment (kushalagrawal1) [inline connectors/resteasy/pom.xml]: Hi @auhlig, I have changed the code but tests are failing. How can I debug this in IntelliJ?

Comment (auhlig) [inline connectors/resteasy/pom.xml]: You could start by enabling debug logging of the http communication `OSFactory.enableHttpLoggingFilter(true);`

Comment (kushalagrawal1) [inline connectors/resteasy/pom.xml]: @auhlig Is it possible to run test cases using intelliJ ?
Do we need to provide any parameter or something ?

Comment (auhlig) [reply connectors/resteasy/pom.xml]: It is. Was a little bit ugly to get this running. You need to to this via maven surefire and attach to this. Search for `maven remote debugging`.

Comment (auhlig) [reply connectors/resteasy/pom.xml]: Were you able to progress here @kushalagrawal1? Else we could merge as-is and release.
Could still go for the resteasy upgrade afterwards.
Ping me if you need help

Comment (kushalagrawal1) [reply connectors/resteasy/pom.xml]: Hi @auhlig, tried setting up IntelliJ but couldn't get succeed. got stuck in regular office tasks.
I can send you git patch if you are ok. I will try to get some time this weekend and finish the task otherwise.

Comment (auhlig) [reply connectors/resteasy/pom.xml]: No worries. I'm also happy to take this in it's current state. Just wanted to know, so do not postpone the release unnecessarily. Many thanks for all the work!
So is it good to merge from your side @kushalagrawal1?

**Comment by kushalagrawal1** [comment]
Hi @auhlig , I tried fixing the issue but I failed.. Here is the patch for the same. Can you please check what is the silly mistake I am doing ?
<change the extension of the file to ".patch">
[issue_1070_patch.txt](https://github.com/ContainX/openstack4j/files/1291381/issue_1070_patch.txt)

File: connectors/resteasy/pom.xml (hunk 1 above)
Comment (aiziyuer) [reply connectors/resteasy/pom.xml]: it works for me

Comment (kushalagrawal1) [reply connectors/resteasy/pom.xml]: @aiziyuer  can you please apply this patch and try?
<change the extension of the file to ".patch">
https://github.com/ContainX/openstack4j/files/1291381/issue_1070_patch.txt

**Comment by kushalagrawal1** [comment]
@auhlig it is ready to merge. can you please review it once.

File: connectors/resteasy/pom.xml (hunk 1 above)
Comment (kushalagrawal1) [reply connectors/resteasy/pom.xml]: Hi @auhlig @vinodborole, I have done all required changes from my side. can you please review it once so it can be merged, or if there is any change need to be done will not affect release.

**Comment by auhlig** [comment]
Very cool @kushalagrawal1 💯 
Any notpicks or objections from your side @vinodborole?

**Comment by vinodborole** [comment]
LGTM @auhlig, good work @kushalagrawal1 👍
